include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,329 @@
+#ifndef _NBL_VIDEO_S_PHYSICAL_DEVICE_LIMITS_H_INCLUDED_
+#define _NBL_VIDEO_S_PHYSICAL_DEVICE_LIMITS_H_INCLUDED_
+
+#include <type_traits>
+#include "nbl/asset/utils/IGLSLCompiler.h" // asset::IGLSLCompiler::E_SPIRV_VERSION
+
+
+namespace nbl::video
+{
+
+struct SPhysicalDeviceLimits
+{
+    /* Vulkan Core 1.0 */
+    uint32_t maxImageDimension1D;
+    uint32_t maxImageDimension2D;
+    uint32_t maxImageDimension3D;
+    uint32_t maxImageDimensionCube;
+    uint32_t maxImageArrayLayers;
+    uint32_t maxBufferViewSizeTexels;
+    uint32_t maxUBOSize;
+    uint32_t maxSSBOSize;
```

Comment:
These are set to `maxUniformBufferRange` and `maxStorageBufferRange ` respectively
and currently `maxBufferSize` is set to `core::max` of these two values.

you suggested that I replace `maxBufferSize` with `Vulkan11Properties::maxMemoryAllocationSize`

but I think there is a misunderstanding here:
> maxUniformBufferRange is the maximum value that can be specified in the range member of a [VkDescriptorBufferInfo](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkDescriptorBufferInfo.html) structure passed to [vkUpdateDescriptorSets](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/vkUpdateDescriptorSets.html) for descriptors of type VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER or VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER_DYNAMIC.

> maxStorageBufferRange is the maximum value that can be specified in the range member of a [VkDescriptorBufferInfo](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkDescriptorBufferInfo.html) structure passed to [vkUpdateDescriptorSets](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/vkUpdateDescriptorSets.html) for descriptors of type VK_DESCRIPTOR_TYPE_STORAGE_BUFFER or VK_DESCRIPTOR_TYPE_STORAGE_BUFFER_DYNAMIC.

they are specifying the range that could be updated to a descriptor set and I don't think they are related to the `maxMemoryAllocationSize` which is:
> maxMemoryAllocationSize is the maximum size of a memory allocation that can be created, even if there is more space available in the heap.

weirdly on some devices including mine `maxStorageBufferRange` is greater than `maxMemoryAllocationSize` 
and for lots of other devices. but bottom line is that they are not equal !
[RTX 3080](https://vulkan.gpuinfo.org/displayreport.php?id=15360#properties_core_10)

Reply:
yeah max UBO, SSBO, TBO is about ranges, so its about sizes of "views" in descriptor bindings.

I believe all my comments about inferring `maxBufferSize` and `maxMemoryAllocationSize` from the above either specifically mentioned that it should only be the case under OpenGL or were made on `COpenGL*` files anyway.

W.r.t. to Vulkan should report both `maxBufferSize` and `maxMemoryAllocationSize ` sizes, if the Vulkan version is less than 1.3 (and no extension like `KHR_maintenance69` existed before) then we should deduce `maxBufferSize=maxMemoryAllocationSize`.